//
// Copyright 2024 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package psi.pir;

enum PirType {
  INVALID = 0;
  SEAL_PIR = 1;
  SPIRAL_PIR = 2;
}

message PirTypeProto {
  PirType type = 1;
}

message CuckooOptionsProto {
  uint64 num_input = 1;
  uint64 num_stash = 2;
  uint64 num_hash = 3;
  double scale_factor = 4;
  uint64 max_try_count = 5;
}

message KwPirOptionsProto {
  CuckooOptionsProto cuckoo_options = 1;
  uint64 key_size = 2;
  uint64 value_size = 3;
}
